### PR TITLE
Mol opt

### PR DESCRIPTION
**Pull request by cpignedoli**
added workchain for slabs, created draft example to run the workchain

### CODE REVIEW

File: aiida_nanotech_empa/workflows/cp2k/protocols/bulk_opt_protocol.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                NGRIDS: 5
+            QS:
+                EPS_DEFAULT: 1.0E-14
+                EXTRAPOLATION: ASPC
+                EXTRAPOLATION_ORDER: 3
+                METHOD: GPW
+            SCF:
+                SCF_GUESS: RESTART
+                EPS_SCF: 1.0E-7
+                MAX_SCF: 20
```
Comment (eimrek) [inline aiida_nanotech_empa/workflows/cp2k/protocols/bulk_opt_protocol.yml:41]: in my experience 20 iterations in OT is not optimal

File: aiida_nanotech_empa/workflows/cp2k/protocols/cell_opt_protocol.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    EACH:
+                        QS_SCF: '0'
+                        GEO_OPT: '0'
+                        CELL_OPT: '0'
+                    ADD_LAST: 'NUMERIC'
+                E_DENSITY_CUBE:
+                    STRIDE: '4 4 4'
+                    EACH:
+                        QS_SCF: '0'
+                        GEO_OPT: '0'
```
Comment (eimrek) [inline aiida_nanotech_empa/workflows/cp2k/protocols/cell_opt_protocol.yml:81]: CELL_OPT missing...

File: aiida_nanotech_empa/workflows/cp2k/cell_opt_workchain.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        super().define(spec)
+        spec.input("code", valid_type=Code)
+        spec.input("structure", valid_type=StructureData)
+
+        spec.input(
+            "charge",  # +1 means one electron removed
+            valid_type=Int,
+            default=lambda: Int(0),
+            required=False)
+        spec.input("fixed_atoms",
```
Comment (eimrek) [inline aiida_nanotech_empa/workflows/cp2k/cell_opt_workchain.py:29]: This argument is not used.

**Review by eimrek — COMMENTED** [review]
One thing to consider Carlo: Would it perhaps make sense to merge the Bulk and Cell opt workchains? They are very similar and there is duplicated code. Did i understand correctly that both support just 3d periodic structures, right? Then maybe it makes sense to just have a single `Cp2kBulkOptWorkChain`, which has a switch for either doing atomic relaxation or cell relaxation. Would probably make maintenance much easier in the future.

**Comment by cpignedoli** [comment]
I fully agree

**Review by eimrek — APPROVED** [review]

**Pull request merged**